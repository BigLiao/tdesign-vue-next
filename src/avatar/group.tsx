import { defineComponent } from 'vue';
import { prefix } from '../config';
import props from './avatar-group-props';
import { TNodeReturnValue } from '../common';
import Avatar from './avatar';
import { renderContent, renderTNodeJSX } from '../utils/render-tnode';

const name = `${prefix}-avatar-group`;

export default defineComponent({
  name: 'TAvatarGroup',
  components: {
    Avatar,
  },

  provide(): Record<string, any> {
    return {
      avatarGroup: this,
    };
  },

  props: {
    ...props,
  },

  methods: {
    renderEllipsisAvatar(children: Array<TNodeReturnValue>): Array<TNodeReturnValue> {
      if (children?.length > this.max) {
        const content = this.setEllipsisContent(children);
        const outAvatar = children.slice(0, this.max);
        outAvatar.push(<Avatar size={this.size} icon={this.isIcon() && typeof this.collapseAvatar !== 'string' ? this.collapseAvatar : null}>{content}</Avatar>);
        return [outAvatar];
      }
      return [children];
    },
    setEllipsisContent(children: Array<TNodeReturnValue>) {
      let content: any = '';
      if (this.collapseAvatar) {
        if (!this.isIcon()) {
          content = renderContent(this, 'collapseAvatar', 'content');
        }
      } else {
        content = `+${children.length - this.max}`;
      }
      return content;
    },
    isIcon() {
      const content = renderTNodeJSX(this, 'collapseAvatar');
      return content;
    },
  },
  render() {
    const { $slots } = this;
    const children: TNodeReturnValue = $slots.default && $slots.default(null);
    const { cascading, max } = this.$props;
    const groupClass = [
      `${name}`,
      {
        [`${prefix}-avatar--offset-right`]: cascading === 'right-up',
        [`${prefix}-avatar--offset-left`]: cascading === 'left-up',
      },
    ];
    let content = [children];

    if (max && max >= 0) {
      content = [this.renderEllipsisAvatar(children)];
    }
    return <div class={groupClass}>
      {content}
    </div>;
  },

});
