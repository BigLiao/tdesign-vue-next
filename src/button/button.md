## Button 按钮

::: demo demos/base 按钮类型
:::

::: demo demos/size 按钮尺寸
:::

::: demo demos/disabled 不可用
:::

::: demo demos/icon 带图标
:::

::: demo demos/round 圆形按钮
:::

::: demo demos/loading loading 状态
:::

### 属性配置
|参数|说明|类型|可选值|默认值|
|--- |--- |--- |--- |--- |
|size|button 大小|string|large/default/small|default|
|theme|button 类型|string|default/primary/text|default|
|round|圆角，如果只有图标，则渲染为圆形按钮|boolean|-|-|
|disabled|禁用|boolean|-|-|
|icon||string|-|-|
|suffixIcon||string|-|-|
|iconOnly|图标按钮|boolean|-|-|
|block|块级按钮|boolean|-|false|
