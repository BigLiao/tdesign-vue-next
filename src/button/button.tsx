import Vue, { CreateElement, VNode } from 'vue';
import { prefix } from '../config';
import RenderComponent from '../utils/render-component';
import CLASSNAMES from '../utils/classnames';
import Icon from '../icon';

const name = `${prefix}-button`;

export default Vue.extend({
  name,
  components: {
    [Icon.name]: Icon,
    RenderComponent,
  },
  props: {
    theme: {
      type: String,
      default: 'line',
      validator(v: string): boolean {
        return (
          [
            'line',
            'primary',
            'dashed',
            'warning',
            'warning-line',
            'link',
            'ghost',
            'ghost-line',
          ].indexOf(v) > -1
        );
      },
    },
    size: {
      type: String,
      default: 'default',
      validator(v: string): boolean {
        return ['large', 'default', 'small'].indexOf(v) > -1;
      },
    },
    icon: [String, Function],
    round: Boolean,
    loading: Boolean,
    block: Boolean,
    disabled: Boolean,
  },
  render(h: CreateElement): VNode { // eslint-disable-line
    const buttonClass = [
      `${name}`,
      CLASSNAMES.SIZE[this.size],
      `${name}--${this.theme}`,
      {
        [CLASSNAMES.STATUS.disabled]: this.disabled,
        [CLASSNAMES.STATUS.loading]: this.loading,
        [`${name}--round`]: this.round,
        [CLASSNAMES.SIZE.block]: this.block,
      },
    ];
    let buttonContent: VNode[] | VNode | string = this.$scopedSlots.default ? this.$scopedSlots.default(null) : '';
    let icon: VNode;

    if (this.loading) {
      icon = <Icon name="loading_gradient"></Icon>;
    } else if (typeof this.icon === 'string') {
      icon = <Icon name={this.icon}></Icon>;
    } else if (typeof icon === 'function') {
      icon = <i class={`${Icon.name}`}>{ this.icon() }</i>;
    }

    if (icon) {
      buttonContent = (
        <span class={`${name}__inner`}>
          { icon }
          {
            buttonContent ? (
              <span class={`${name}__text`}>
                { buttonContent }
              </span>
            ) : ''
          }
        </span>
      );
    }

    return (
      <button class={buttonClass} { ...{ on: this.$listeners } }>
        { 
          
         }
      </button>
    );
  },
});
