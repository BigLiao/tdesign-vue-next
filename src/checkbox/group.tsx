import { defineComponent, h } from 'vue';
import { prefix } from '../config';
import Checkbox from './checkbox';
import checkboxGroupProps from '@TdTypes/checkbox-group/props';
import { CheckboxOptionObj, TdCheckboxProps, CheckboxGroupValue } from '@TdTypes/checkbox/TdCheckboxProps';
import intersection from 'lodash/intersection';

const name = `${prefix}-checkbox-group`;

export default defineComponent({
  name,
  components: {
    Checkbox,
  },
  emits: ['change'],
  props: { ...checkboxGroupProps },

  data() {
    return {
      checkedMap: {},
    };
  },

  watch: {
    values: {
      immediate: true,
      handler() {
        if (this.value instanceof Array) {
          const map = {};
          this.value.forEach((item: string | number) => {
            map[item] = true;
          });
          this.checkedMap = map;
        }
      },
    },
  },

  computed: {
    optionList(): Array<CheckboxOptionObj> {
      if (!this.options) return [];
      return this.options.map((item) => {
        let r: CheckboxOptionObj = {};
        if (typeof item !== 'object') {
          r = { label: String(item), value: item } ;
        } else {
          r = { ...item };
          r.disabled = r.disabled === undefined ? this.disabled : r.disabled;
        }
        return r;
      });
    },
    values(): string {
      if (this.value instanceof Array) {
        return this.value.join();
      }
      return '';
    },
    intersectionLen(): number {
      const values = this.optionList.map(item => item.value);
      if (this.value instanceof Array) {
        const n = intersection(this.value, values);
        return n.length;
      }
      return 0;
    },
    isCheckAll(): boolean {
      if (this.value instanceof Array && this.value.length !== this.optionList.length - 1) {
        return false;
      }
      return this.intersectionLen === this.optionList.length - 1;
    },
    indeterminate(): boolean {
      return !this.isCheckAll && this.intersectionLen < this.optionList.length && this.intersectionLen !== 0;
    },
  },

  render() {
    return (
      <div class={name}>
        {!!this.optionList.length && this.optionList.map((option, index) => {
          if (option.checkAll) return this.renderCheckAll(option);
          return (
            <Checkbox
              key={index}
              props={{ ...option }}
              checked={this.checkedMap[option.value]}
            >
              {this.renderLabel(option)}
            </Checkbox>
          );
        })}
        {this.$slots.default() && this.$slots.default(null)}
      </div>
    );
  },

  methods: {
    renderCheckAll(option: CheckboxOptionObj) {
      return (
        <Checkbox
          checked={this.isCheckAll}
          indeterminate={this.indeterminate}
          onChange={this.onCheckAllChange}
          data-name='TDESIGN_CHECK_ALL'
          props={{ ...option }}
        >{this.renderLabel(option)}</Checkbox>
      );
    },
    renderLabel(option: CheckboxOptionObj) {
      if (typeof option.label === 'function') {
        return option.label(h);
      }
      return option.label;
    },
    emitChange(val: CheckboxGroupValue, e?: Event) {
      this.$emit('change', val, { e });
    },
    handleCheckboxChange(data: { checked: boolean; e: Event; option: TdCheckboxProps }) {
      const oValue = data.option.value;
      if (this.value instanceof Array) {
        const val = [...this.value];
        if (data.checked) {
          val.push(oValue);
        } else {
          const i = val.indexOf(oValue);
          val.splice(i, 1);
        }
        this.emitChange(val, data.e);
        this.setCheckedMap(data.option.value, data.checked);
      }
    },
    setCheckedMap(value: string | number, checked: boolean) {
      this.checkedMap[value] = checked;
    },
    onCheckAllChange(checked: boolean, context: { e: Event }) {
      if (checked) {
        const val = [];
        for (let i = 0, len = this.optionList.length; i < len; i++) {
          const item = this.optionList[i];
          if (item.checkAll) continue;
          val.push(item.value);
          this.checkedMap[item.value] = true;
        }
        this.emitChange(val, context.e);
      } else {
        this.emitChange([], context.e);
        this.checkedMap = {};
      }
    },
  },
});