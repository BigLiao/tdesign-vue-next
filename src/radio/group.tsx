import { VNode, defineComponent, h } from 'vue';
import isString from 'lodash/isString';
import isNumber from 'lodash/isNumber';
import props from '../../types/radio-group/props';
import { RadioOptionObj, RadioOption, RadioValue } from '../../types/radio/TdRadioProps';
import { prefix } from '../config';
import Radio from './radio';
import { TNodeReturnValue } from '../common';

const name = `${prefix}-radio-group`;

export default defineComponent({
  name,
  components: {
    Radio,
  },
  provide() {
    return {
      radioGroup: this,
    };
  },
  props: { ...props },
  emits: ['change'],
  methods: {
    handleRadioChange(value: RadioValue, context: { e: Event }) {
      this.$emit('change', value, context);
    },
  },
  render(): VNode {
    const { $slots } = this;
    let children: TNodeReturnValue = $slots.default && $slots.default(null);

    if (this.options && this.options.length) {
      children = (this.options).map((option: RadioOption) => {
        let opt = option as RadioOptionObj;
        if (isNumber(option) || isString(option)) {
          opt = { value: option, label: option.toString() };
        }
        return (
          <Radio
            key={`radio-group-options-${opt.value}-${Math.random()}`}
            name={this.name}
            checked={this.value === opt.value}
            disabled={'disabled' in opt ? opt.disabled : this.disabled}
            value={opt.value}
          >
            {typeof opt.label === 'function' ? opt.label(h) : opt.label}
          </Radio>
        );
      });
    }

    const groupClass = [
      `${name}`,
      `${name}-${this.buttonStyle}`,
      `${name}-${this.size}`,
    ];

    return (
      <div class={groupClass}>
        {children}
      </div>
    );
  },
});
