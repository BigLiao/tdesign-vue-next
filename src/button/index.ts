import _Button from './button';
import { TdButtonProps } from '@TdTypes/button/TdButtonProps';
import { withInstall, WithInstallType } from '../utils/withInstall';

const Button: WithInstallType<typeof _Button> = withInstall(_Button);

export type ButtonProps = TdButtonProps;
export { Button };
export default Button;
