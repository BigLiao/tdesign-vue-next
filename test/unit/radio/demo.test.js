import { mount } from '@vue/test-utils';
import base from '@/examples/radio/demos/base.vue';
import group from '@/examples/radio/demos/group.vue';
import button from '@/examples/radio/demos/button.vue';
import size from '@/examples/radio/demos/size.vue';

// unit test for component in examples.
describe('Radio', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('radio group demo works fine', () => {
    const wrapper = mount(group);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('radio button demo works fine', () => {
    const wrapper = mount(button);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('radio button size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
});

