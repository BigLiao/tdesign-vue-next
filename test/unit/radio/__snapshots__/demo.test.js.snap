// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio base demo works fine 1`] = `
<div
  class="tdesign-radio-base"
>
  <div
    class="tdesign-demo-block"
  >
    <label
      class="t-radio"
      modelvalue="false"
    >
      <input
        class="t-radio__former"
        name=""
        type="radio"
      />
      <span
        class="t-radio__input"
      />
      <span
        class="t-radio__label"
      >
        
        未选中项
        
      </span>
    </label>
    <label
      class="t-radio t-is-checked"
      modelvalue="false"
    >
      <input
        class="t-radio__former"
        name=""
        type="radio"
      />
      <span
        class="t-radio__input"
      />
      <span
        class="t-radio__label"
      >
        
        选中项
        
      </span>
    </label>
    <label
      class="t-radio t-is-checked t-is-disabled"
      modelvalue="false"
    >
      <input
        class="t-radio__former"
        disabled=""
        name=""
        type="radio"
      />
      <span
        class="t-radio__input"
      />
      <span
        class="t-radio__label"
      >
        
        选中禁用态
        
      </span>
    </label>
    <label
      class="t-radio t-is-disabled"
      modelvalue="false"
    >
      <input
        class="t-radio__former"
        disabled=""
        name=""
        type="radio"
      />
      <span
        class="t-radio__input"
      />
      <span
        class="t-radio__label"
      >
        
        未选中禁用态
        
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio radio button demo works fine 1`] = `
<div
  class="tdesign-radio-button"
>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-outline"
    >
      
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="1"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项一
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="3"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项二
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked t-is-disabled"
      >
        <input
          class="t-radio-button__former"
          disabled=""
          name=""
          type="radio"
          value="3"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项三
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-disabled"
      >
        <input
          class="t-radio-button__former"
          disabled=""
          name=""
          type="radio"
          value="4"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项四
          
          
        </span>
      </label>
      
    </div>
  </div>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-filled"
    >
      
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="bj"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项一
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="gz"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项二
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked t-is-disabled"
      >
        <input
          class="t-radio-button__former"
          disabled=""
          name=""
          type="radio"
          value="gz"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项三
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-disabled"
      >
        <input
          class="t-radio-button__former"
          disabled=""
          name=""
          type="radio"
          value="sz"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
          选项四
          
          
        </span>
      </label>
      <div
        class="t-radio-group-filled-bg-block"
      />
      
    </div>
  </div>
</div>
`;

exports[`Radio radio button size demo works fine 1`] = `
<div
  class="tdesign-demo-block-row"
>
  <div
    class="tdesign-demo-block-column"
  >
    <div>
      <div
        class="t-radio-group t-radio-group-small t-radio-group-outline"
      >
        
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        
      </div>
    </div>
    <div>
      <div
        class="t-radio-group t-radio-group-medium t-radio-group-outline"
      >
        
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        
      </div>
    </div>
    <div>
      <div
        class="t-radio-group t-radio-group-large t-radio-group-outline"
      >
        
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        
      </div>
    </div>
  </div>
  <div
    class="tdesign-demo-block-column"
  >
    <div>
      <div
        class="t-radio-group t-radio-group-small t-radio-group-filled"
      >
        
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        <div
          class="t-radio-group-filled-bg-block"
        />
        
      </div>
    </div>
    <div>
      <div
        class="t-radio-group t-radio-group-medium t-radio-group-filled"
      >
        
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        <div
          class="t-radio-group-filled-bg-block"
        />
        
      </div>
    </div>
    <div>
      <div
        class="t-radio-group t-radio-group-large t-radio-group-filled"
      >
        
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bj"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            北京
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sh"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            上海
            
            
          </span>
        </label>
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="gz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            广州
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="sz"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
            深圳
            
            
          </span>
        </label>
        <div
          class="t-radio-group-filled-bg-block"
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Radio radio group demo works fine 1`] = `
<div
  class="tdesign-radio-group"
>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-outline"
      modelvalue=""
    >
      
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name="city"
          type="radio"
          value="bj"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          北京
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name="city"
          type="radio"
          value="sh"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          上海
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name="city"
          type="radio"
          value="gz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          广州
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name="city"
          type="radio"
          value="sz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          深圳
          
        </span>
      </label>
      
    </div>
  </div>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-outline"
    >
      
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="bj"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          北京
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="sh"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          上海
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="gz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          广州
          
        </span>
      </label>
      <label
        class="t-radio t-is-checked"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="sz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          深圳
          
        </span>
      </label>
      
    </div>
  </div>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-outline"
    >
      
      <label
        class="t-radio t-is-disabled"
      >
        <input
          class="t-radio__former"
          disabled=""
          name=""
          type="radio"
          value="bj"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          北京
          
        </span>
      </label>
      <label
        class="t-radio t-is-disabled"
      >
        <input
          class="t-radio__former"
          disabled=""
          name=""
          type="radio"
          value="sh"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          上海
          
        </span>
      </label>
      <label
        class="t-radio t-is-checked"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="gz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          广州
          
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="sz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          深圳
          
        </span>
      </label>
      
    </div>
  </div>
  <div>
    <div
      class="t-radio-group t-radio-group-medium t-radio-group-outline"
    >
      
      <label
        class="t-radio"
        modelvalue="false"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="bj"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          北京
          
        </span>
      </label>
      <label
        class="t-radio"
        modelvalue="false"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="sh"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          上海
          
        </span>
      </label>
      <label
        class="t-radio t-is-checked"
        modelvalue="false"
      >
        <input
          class="t-radio__former"
          name=""
          type="radio"
          value="gz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          广州
          
        </span>
      </label>
      <label
        class="t-radio t-is-disabled"
        modelvalue="false"
      >
        <input
          class="t-radio__former"
          disabled=""
          name=""
          type="radio"
          value="sz"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          
          深圳
          
        </span>
      </label>
      
    </div>
  </div>
</div>
`;
